record/record_test.go

```diff
@@ -314,10 +314,25 @@ func TestStaleWriter(t *testing.T) {
 	}
 }
 
-func writeRecords(records [][]byte) ([]byte, error) {
+type testRecords struct {
+	records [][]byte // The raw value of each record.
+	buf     []byte   // The serialized records form of all records.
+}
+
+// makeTestRecords generates test records of specified lengths. The raw
+// records are returned as *testRecords. The first record will consist of
+// repeating 0x00 bytes, the next record of 0x01 bytes, and so forth. The
+// values will loop back to 0x00 after 0xff.
+func makeTestRecords(recordLengths ...int) (*testRecords, error) {
+	results := &testRecords{}
```

Comment:
Just a personal preference, but I like to name whatever will become the return value "ret".